gem/lib/metasploit-payloads.rb

```diff
@@ -10,48 +12,126 @@ module MetasploitPayloads
   METERPRETER_SUBFOLDER = 'meterpreter'
   USER_DATA_SUBFOLDER   = 'payloads'
 
+  #
+  # @return [Array<Hash<String, Symbol>>] An array of filenames with warnings. Provides a file name and reason.
+  #  Empty if all needed Meterpreter files exist and have the correct hash.
+  def self.file_warnings
+    file_warnings = []
+
+    begin
+      manifest_contents = ::File.binread(manifest_path)
+    rescue ::Errno::ENOENT => _e
+      return [{ path: 'manifest', reason: :not_found }]
```

Comment:
Can we update the paths here to always be absolute, as it's hard from a caller's perspective to know which type of value this function is returning